src/timer.rs

```diff
@@ -58,12 +59,74 @@ pub mod config {
         fn default() -> Self {
             Self {
                 divider: 80,
-                #[cfg(any(esp32s2, esp32s3, esp32c3))]
+                #[cfg(not(esp32))]
                 xtal: false,
                 auto_reload: false,
             }
         }
     }
+
+    #[cfg(not(esp_idf_version_major = "4"))]
+    #[allow(clippy::upper_case_acronyms)]
+    pub(crate) enum ClockSource {
+        #[cfg(any(esp32, esp32s2, esp32s3, esp32c3))]
+        APB,
+        #[cfg(esp32c2)]
+        PLL40,
+        #[cfg(esp32h2)]
+        PLL48,
+        #[cfg(esp32c6)]
+        PLL80,
+        #[cfg(not(esp32))]
+        XTAL,
+    }
+
+    #[cfg(not(esp_idf_version_major = "4"))]
+    impl Default for ClockSource {
```

Comment:
Nit: you can replace all of this with just `#[default]` above, because for each MCU you end up with exactly two members of the enum, where the `XTAL` member should always be the non-default one.

Reply:
done